src/java.base/share/classes/jdk/internal/foreign/AbstractMemorySegmentImpl.java

```diff
@@ -505,7 +505,7 @@ public int characteristics() {
 
     @Override
     public String toString() {
-        return "MemorySegment{ heapBase: " + heapBase() + " address:" + address() + " limit: " + length + " }";
+        return "MemorySegment{ heapBase: " + heapBase() + " address: 0x" + Long.toHexString(address()) + " limit: " + length + " }";
```

Comment:
should we replace `limit` with `size` since we're here? (the javadoc says segments have size, not limit)